packages/jest-runtime/src/index.js

```diff
@@ -267,40 +270,55 @@ class Runtime {
     return cliOptions;
   }
 
+  teardown() {
+    // $FlowFixMe: de-reference environment.
+    this._environment = null;
+  }
+
   requireModule(
     from: Path,
     moduleName?: string,
     options: ?InternalModuleOptions,
   ) {
-    const moduleID = this._resolver.getModuleID(
-      this._virtualMocks,
-      from,
-      moduleName,
-    );
+    const isNativeModule =
+      moduleName &&
+      ((options && options.isNativeModule) ||
+        this._resolver.isCoreModule(moduleName));
+
+    const moduleID =
+      !isNativeModule &&
+      this._resolver.getModuleID(this._virtualMocks, from, moduleName);
+
+    let moduleRegistry;
     let modulePath;
 
-    const moduleRegistry =
-      !options || !options.isInternalModule
-        ? this._moduleRegistry
-        : this._internalModuleRegistry;
+    if (isNativeModule) {
+      moduleRegistry = this._nativeModuleRegistry;
+    } else if (options && options.isInternalModule) {
+      moduleRegistry = this._internalModuleRegistry;
```

Comment:
The internal module registry is used in Jest for Jest's own core modules like `pretty-format` or `chalk`. Any reason you can't reuse that one? Why do you need a third one?

Reply:
Because of clashes. Once you start requiring inside a Jest internal, all the child modules are required inside the Jest internal module as well (which is fine, because you prevent clashes between test and Jest code).

If Jest requires `internal/util`, `internal` refers to NPM's module; but if you require it from the Native `util` module, it refers to the internal file; so you need a separate `require` implementation (as well as a separate registry) to avoid the clash.

Reply:
Ok, that makes sense.